### PR TITLE
Dockerize app

### PR DESCRIPTION
**Pull request by ezerw**
I was about to start trying to help with other of the reported issues but didn't have `MySQL`, `PHP` or `Supervisord` installed locally so I thought will be a good idea to dockerize the app first and then continue with next issue.

In addition to the services needed to serve the application, I added a `Makefile` to start, stop and login into the services but also to run some of the extra steps to configure the app (copy .env.example to .env if the file doesn't exist and run `cmd.php` to start the websocket server)

- To start the application in docker containers run `make up` on the root directory.
- To stop all services run `make down` on the root directory.
- To `SSH` into the `app` container run `make login`

## Manual tests
- [x] Start all services (PHP-FPM, Nginx and MySQL) and open the application in the browser.
- [x] Register successfully a new user.
- [x] Confirmed the websocket service is running.

### CODE REVIEW

**Comment by ankitjain28may** [comment]
@ezerw Looks like web socket is not working, as you haven't executed the cmd.php file anywhere in the docker, how will it work.

**Comment by ezerw** [comment]
> @ezerw Looks like web socket is not working, as you haven't executed the cmd.php file anywhere in the docker, how will it work.

Hey @ankitjain28may , as mentioned in the PR description one of the things that are executed when you run `make up` is `docker-compose exec -d app php cmd.php`. That will execute what is in `cmd.php` using php-cli inside the app container.

As I understand the order will be:
- Execute `cmd.php` that will start the websocket server exposed on port 8080
- Message.js file should be modified to connect the client to the same IP in that port
Am I correct? 

What is the expected behaviour visiting `http://127.0.0.1:8888/views/message.php`?

**Comment by ezerw** [comment]
I've replaced docker-compose service with programs running in supervisor.
Also did some other changes to get the websocket client <> server connection and it worked but because of other errors e.g:
```
Fatal error: Uncaught Error: Call to a member function getCookies() on null in /var/www/html/src/Chat.php:83
```
```
oc-app | New connection! (86)
oc-app | Failed
oc-app | Warning: count(): Parameter must be an array or an object that implements Countable in /var/www/html/src/Receiver.php on line 80
oc-app |
oc-app | Warning: Creating default object from empty value in /var/www/html/src/Receiver.php on line 85
oc-app | Connection 86 has disconnected
```
The connection fails saying that the response format is not correct and other messages like that. It should be because of PHP versions (Docker container using 7.3) or other bugs but as far as dockerize the application this does it.

Once this is merged I could give you a hand debugging and fixing errors but they should be treated as different issues as are not related to docker.

File: .dockerignore [hunk 1]
```diff
@@ -0,0 +1,9 @@
+**/*.log
+**/._*
+**/.DS_Store
+**/.gitignore
+**/.gitattributes
+**/Thumbs.db
+.dockerignore
+Dockerfile
+docker-compose.yaml
```
Comment (ankitjain28may) [inline .dockerignore]: end line

File: .env.example [hunk 2]
```diff
@@ -1,11 +1,11 @@
...
-DB_PORT = 3306
-DB_NAME = openchat
-DB_USER = root
-DB_PASSWORD = password
+DB_CONNECTION=mysql
+DB_HOST=db
+DB_PORT=3306
+DB_NAME=openchat
+DB_USER=user
+DB_PASSWORD=password
```
Comment (ankitjain28may) [inline .env.example]: Add DB_ROOT_PASSWORD

File: docker-compose.yml [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+  # ========================
+  db:
+    image: mysql:5.7
+    container_name: oc-db
+    restart: unless-stopped
+    ports:
+      - 33060:3306
+    environment:
+      # `secret` as password is used only in local development.
+      MYSQL_ROOT_PASSWORD: secret
```
Comment (ankitjain28may) [inline docker-compose.yml]: Add to env variables

File: docker-compose.yml [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+      - vendor:/var/www/html/vendor
+      - node_modules:/var/www/html/node_modules
+    depends_on:
+      - db
+    working_dir: /var/www/html
+    ports:
+      - 8080:8080
+      - 8888:8888
+    networks:
+      - openchat
```
Comment (ankitjain28may) [inline docker-compose.yml]: end line

File: docker/config/nginx/default.conf [hunk 5]
```diff
@@ -0,0 +1,32 @@
+server {
```
Comment (ankitjain28may) [inline docker/config/nginx/default.conf]: It's not required now.

File: docker/config/php/php.ini [hunk 6]
```diff
@@ -0,0 +1,3 @@
+upload_max_filesize = 10M
+post_max_size = 10M
+memory_limit = 128M
```
Comment (ankitjain28may) [inline docker/config/php/php.ini]: end line

File: docker/config/supervisor/supervisord.conf [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+[program:websockets]
+user=root
+directory=/var/www/html
+command=php cmd.php
+
+stdout_logfile=/dev/fd/1
+stdout_logfile_maxbytes=0
+
+stderr_logfile=/dev/fd/2
+stderr_logfile_maxbytes=0
```
Comment (ankitjain28may) [inline docker/config/supervisor/supervisord.conf]: end line

**Comment by ankitjain28may** [comment]
@ezerw Your previous commit is fine too, rather defining the `php cmd.php` as a command, we can define it as an entry point and make `nginx` container dependent on `php` container so whenever nginx runs, it first executes the `php` container and associated WebSocket service.

Moreover, as it is breaking the code, let's not push to master, push the code to the `docker` branch.

**Comment by ezerw** [comment]
> @ezerw Your previous commit is fine too, rather defining the `php cmd.php` as a command, we can define it as an entry point and make `nginx` container dependent on `php` container so whenever nginx runs, it first executes the `php` container and associated WebSocket service.
> 
> Moreover, as it is breaking the code, let's not push to master, push the code to the `dockerize` branch.

@ankitjain28may I've pushed the requested changes, let me know when this PR is merged.

**Pull request merged**